### ISSUE TITLE
Track down possible impacts on Jax auto gradient functions from jnp.where

### ISSUE THREAD

**Issue by arik-shurygin** [CDCgov/DynODE#120]
Currently the model uses jnp.where in a couple of key places instead of using If branches. The issue with jnp.where with regards to autograd is described in the [docs](https://jax.readthedocs.io/en/latest/_autosummary/jax.numpy.where.html):

 _"Special care is needed when the x or y input to [jax.numpy.where()](https://jax.readthedocs.io/en/latest/_autosummary/jax.numpy.where.html#jax.numpy.where) could have a value of NaN. Specifically, when a gradient is taken with [jax.grad()](https://jax.readthedocs.io/en/latest/_autosummary/jax.grad.html#jax.grad) (reverse-mode differentiation), a NaN in either x or y will propagate into the gradient, regardless of the value of condition. More information on this behavior and workarounds is available in the JAX FAQ:"_

Start by looking at abstract_parameters.py `seasonality()` for a clear use of jnp.where with possible impacts on auto_grad().

**Comment by arik-shurygin** [comment]
When thinking about this deeper this issue pertains specifically to when the values in `x` or `y` are NaN. In our use case of `jnp.where` we are aiming to specifically avoid NaNs in the output by diverting the gradient away from NaN values. So this issue does not come up except in rare edge cases where inference will likely be failing regardless (diverging chains etc)
